go-fuzz-dep/main.go

```diff
@@ -21,97 +17,24 @@ import (
 type Bool = bool
 
 var (
-	inFD  FD
-	outFD FD
-
 	CoverTab    *[CoverSize]byte
-	input       []byte
 	sonarRegion []byte
 	sonarPos    uint32
+	CoverTabTmp [CoverSize]byte
 )
 
-func init() {
-	var mem []byte
-	mem, inFD, outFD = setupCommFile()
-	CoverTab = (*[CoverSize]byte)(unsafe.Pointer(&mem[0]))
-	input = mem[CoverSize : CoverSize+MaxInputSize]
-	sonarRegion = mem[CoverSize+MaxInputSize:]
-}
 
-func Main(f func([]byte) int) {
-	runtime.GOMAXPROCS(1) // makes coverage more deterministic, we parallelize on higher level
```

Comment:
If this is going to get upstreamed, we obviously need to find some way not to delete all this code, but instead have it live harmoniously side-by-side. :)

It might even make sense just to make a new package instead, and have go-fuzz-build switch as needed; it doesn't seem like there's much code overlap.

Reply:
Could build tags be of any help here? So that if go-fuzz-build runs the compilation process, the tag 'libFuzzer' would use a special version of ```go-fuzz-dep/main.go````, without a build tag it uses the default?
IDK, I'm not really a Go guy.

Reply:
Yes, we could use build tags. It depends on how much code overlap there is. If there's substantive code overlap, build tags would be a fine idea. If the code is mostly disjoint, we may as well just have a go-fuzz-dep-libfuzzer instead.

Reply:
Let's do 2 files and build tags. go-fuzz-dep is mentioned a bunch of times in go-fuzz-build, so just adding a tag looks simpler. Tag may turn out to be useful in other places as well (maybe even in user code?).
And in the end this is all strictly internal to go-fuzz, so we can change it in future.
Since we currently define `gofuzz`, let's go `gofuzz-libfuzzer` (or `gofuzz_libfuzzer` id dashes are not allowed) to keep all our tags prefixed with gofuzz.

Reply:
Now supporting 2 files using build tags.